### DISCUSSION TITLE
Function not working in Seurat5 because of layers

### DISCUSSION THREAD

**Discussion by MarcElvld** [satijalab/seurat#8047] [Q&A]
Dear all,

Working with a merged file of 6 samples in Seurat, as a QC I want to make a list of the genes with the highest read number per cell : 
Seu$largest_count <- apply(Seu[['RNA']]@counts,2, max) #Return per cell highest read number
Seu$largest_index <- apply(Seu[['RNA']]@counts,2, which.max) #Return rownumber max reads
Seu$largest_gene <- rownames(Seu)[Seu$largest_index] #Return row names of genes with max reads

But in Seurat5 the different samples are stored in different layers. So when I run this code (with the @ replaced by $) I only get numbers from the first layer. How can I make the function to run through the counts of all layers?

Thank you,
Marc

**Comment by camilletestard** [comment]
Dear all, 

I am experiencing the same issue, hopefully bumping up the request! Can we make the "apply" function to run through the counts of all layers?

Thanks!

**Comment by wudustan** [comment]
You could replace `Seu` with `JoinLayers(Seu)`
